omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,214 @@
+	public String getDisplayString(IdentifierSource identifierSource) {
+		return identifierSource.getIdentifierType() + " - " + identifierSource.getName();
+	}
+	
+	@Override
+	public IdentifierSource newDelegate() {
+		return new SequentialIdentifierGenerator();
+	}
+	
+	@Override
+    public Object create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		
+		Object savedIdentifierSource = null;
+		String identifierSourceType = postBody.get("sourceType").toString();
+		
+		if(identifierSourceType.equals("SequentialIdentifierGenerator")) {
+			
+			SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+			
+			String identifierTypeUuid = postBody.get("identifierType").toString();
+			String name = postBody.get("name").toString();
+			String description = postBody.get("description").toString();
+			String firstIdentifierBase = postBody.get("firstIdentifierBase").toString();
+			String baseCharacterSet = postBody.get("baseCharacterSet").toString();
+			String prefix = postBody.get("prefix").toString();
+			String suffix = postBody.get("suffix").toString();
+			Integer minLength = Integer.parseInt(postBody.get("minLength").toString());
+			Integer maxLength = Integer.parseInt(postBody.get("maxLength").toString());
+			
+			identifierSource.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid));
+			identifierSource.setName(name);
+			identifierSource.setDescription(description);
+			identifierSource.setBaseCharacterSet(baseCharacterSet);
+			identifierSource.setFirstIdentifierBase(firstIdentifierBase);
+			identifierSource.setPrefix(prefix);
+			identifierSource.setSuffix(suffix);
+			identifierSource.setMinLength(minLength);
+			identifierSource.setMaxLength(maxLength);
+			
+			savedIdentifierSource =  save(identifierSource);	
+		}
+		else if(identifierSourceType.equals("RemoteIdentifierSource")) {
+			
+			RemoteIdentifierSource identifierSource = new RemoteIdentifierSource();
+			
+			String identifierTypeUuid = postBody.get("identifierType").toString();
+			String name = postBody.get("name").toString();
+			String description = postBody.get("description").toString();
+			String username = postBody.get("username").toString();
+			String password = postBody.get("password").toString();
+			String url = postBody.get("url").toString();
+			
+			identifierSource.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid));
+			identifierSource.setName(name);
+			identifierSource.setUrl(url);
+			identifierSource.setDescription(description);
+			identifierSource.setUser(username);
+			identifierSource.setPassword(password);
+			
+			savedIdentifierSource =  save(identifierSource);
+			
+		}
+		else if(identifierSourceType.equals("IdentifierPool")){
+			
+			IdentifierPool identifierSource = new IdentifierPool();
+			
+			String identifierTypeUuid = postBody.get("identifierType").toString();
```

Comment:
Can you also avoid repeating postBody.get("identifierType").toString(); three times? The same applies for any other property that you are getting more than once.

Reply:
fixed.